### PR TITLE
Remove redundant NULL buffer checks

### PR DESCRIPTION
**Pull request by vyivel**
From what I understand, (confusingly named) `surface_apply_damage()` doesn't update `wlr_surface.buffer` if the upload has failed.

### CODE REVIEW

File: include/wlr/types/wlr_surface.h [hunk 1]
```diff
@@ -91,8 +91,8 @@ struct wlr_surface {
 	/**
 	 * The surface's buffer, if any. A surface has an attached buffer when it
 	 * commits with a non-null buffer in its pending state. A surface will not
-	 * have a buffer if it has never committed one, has committed a null buffer,
-	 * or something went wrong with uploading the buffer.
```
Comment (emersion) [inline include/wlr/types/wlr_surface.h:95]: If the first attached buffer failed during import/upload, this can still happen.

Comment (vyivel) [reply include/wlr/types/wlr_surface.h:95]: Right. Although, if that were to happen, the surface would've been unmapped already, so there's no reason to have those checks anyway.
Should I change it to "something went wrong with uploading the *first* buffer"?

File: types/wlr_layer_shell_v1.c [hunk 2]
```diff
@@ -339,10 +339,6 @@ static void layer_surface_role_commit(struct wlr_surface *wlr_surface) {
 		surface->mapped = true;
 		wlr_signal_emit_safe(&surface->events.map, surface);
 	}
-	if (surface->configured && !wlr_surface_has_buffer(surface->surface) &&
```
Comment (emersion) [inline types/wlr_layer_shell_v1.c:342]: This is not done in the precommit handler for layer-shell.

Comment (vyivel) [reply types/wlr_layer_shell_v1.c:342]: Hm. It should be, though.

Comment (vyivel) [reply types/wlr_layer_shell_v1.c:342]: Added role precommit handler for `layer-surface`.

**Review by emersion — APPROVED** [review]
LGTM, thanks!

**Pull request merged**